src/App.js

```diff
@@ -1,80 +1,67 @@
+import ErrorFallback from "./common/ErrorFallback";
+import infoIcon from "./svg/Info.svg";
+import { useQueryClient } from "@tanstack/react-query";
+import { keys } from "./api/keys";
 
-/** UGC Analyzer application
+/** Who Talked About Us App
  *
  * - indexes: list of indexes and loading status
- *   { data: [{_id: '1', index_name: 'testIndex2', index_options: Array(4),...},
- *            {_id: '2', index_name: 'testIndex2', index_options: Array(4),...}]
- *          , isLoading: false }
  *
  * App -> { IndexForm, VideoIndex }
  */
 
 function App() {
-  const [indexes, setIndexes] = useState({
-    data: null,
-    isLoading: true,
-  });
+  const queryClient = useQueryClient();
+  const { data: indexesData, refetch } = useGetIndexes();
+  const indexes = indexesData?.data.data;
+  
+  useEffect(() => {
+    queryClient.invalidateQueries({ queryKey: [keys.INDEXES] });
+  }, [indexes]);
 
-  useEffect(function fetchIndexesOnMount() {
-    async function fetchIndexes() {
-      const response = await TwelveLabsApi.getIndexes();
-      setIndexes({ data: response, isLoading: false });
-    }
-    fetchIndexes();
-  }, []);
-
-  /** Triggered by index form submit; creates/adds a new index */
-  async function addIndex(indexName) {
-    const newIndex = await TwelveLabsApi.createIndex(indexName);
-    setIndexes((indexes) => ({
-      data: [{ ...newIndex, index_name: indexName }, ...indexes.data],
-      isLoading: false,
-    }));
-  }
-
-  if (indexes.isLoading)
-    return (
-      <div className="text-center">
-        <div className=" loading-spinner">
-          <img src={loadingSpinner} alt="Loading Spinner" />
-        </div>
-        Loading
-      </div>
-    );
   return (
     <div className="App">
       <Container className="m-auto p-3">
         <h1 className="m-3 display-5">Who Talked About Us?</h1>
         <h4>Find the right influencers (organic brand fans) to reach out </h4>
       </Container>
       <Container className="m-auto p-3 indexFormContainer">
-        <IndexForm indexes={indexes.data} addIndex={addIndex} />
-      </Container>
-      <Container className="m-auto p-3">
-        {indexes.data &&
-          indexes.data.map((index) => (
-            <div className="mb-3" key={index._id}>
-              <VideoIndex
-                indexes={indexes}
-                setIndexes={setIndexes}
-                index={index}
-                key={index._id}
-                closeIcon={closeIcon}
-                backIcon={backIcon}
-                loadingSpinner={loadingSpinner}
-              />
-            </div>
-          ))}
+        <IndexForm />
       </Container>
+      {indexes?.length === 0 && (
+        <div className="doNotLeaveMessageWrapper">
+          <img src={infoIcon} alt="infoIcon" className="icon"></img>
+          <div className="doNotLeaveMessage">
+            There is no index. Start creating one!
+          </div>
+        </div>
+      )}
+      {indexes?.length > 0 && (
```

Comment:
Since `App` component is not wrapped with ErrorBoundary, occasionally error could be happened and type of `indexes?.length` could be `undefined`. If it happens, application will break.

So instead of comparing number directly with optional variable, you need to wrap this condition by checking if `indexes` is not undefined value.

```suggestion
      {indexes != null && indexes.length > 0 && (
```

But better way is handling errors before rendering this section